vendor/github.com/portworx/sched-ops/k8s/k8s.go

```diff
@@ -930,6 +934,28 @@ func (k *k8sOps) DescribeDeployment(depName, depNamespace string) (*apps_api.Dep
 	return &dep.Status, err
 }
 
+func (k *k8sOps) UpdateDeployment(deployment *apps_api.Deployment, newRepl *int32) error {
+	logrus.Info("Attempt to update the deployment.")
+	dep, err := k.appsClient().Deployments(deployment.Namespace).Get(deployment.Name, meta_v1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	*dep.Spec.Replicas = *newRepl
+	_, err = k.appsClient().Deployments(dep.Namespace).Update(dep)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k *k8sOps) GetScaleFactor(deployment *apps_api.Deployment) (int32, error) {
```

Comment:
I don't think we need this method.. the caller can always do a GetDeployment, change the replicaSet and call update..

Reply:
New method GetNewScaleFactor in scheduler/k8s/k8s.go